.github/workflows/deploy.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
```

Comment:
refactor into 2 jobs:
1. build and store build files into cache (also can restore cache if that speeds it up, but storing the build will allow you to inspect it)
2. load the cached build and deploy it (on main branch only)

Reply:
onto this!